src/hooks/useNotificationSound.ts

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useState } from "react";
+import useSound from "use-sound";
+
+import notificationSound from "assets/sounds/newmessage.m4a";
+
+import { PrivateChatMessage, ChatTypes } from "types/chat";
+
+import { useChatSidebarControls } from "hooks/chatSidebar";
+
+import { useUser } from "./useUser";
+import { WithId } from "utils/id";
+
+export const useNotificationSound = (
+  privateChatMessages?: WithId<PrivateChatMessage>[]
+) => {
+  const { user } = useUser();
+  const userId = user?.uid;
+
+  const { chatSettings } = useChatSidebarControls();
+
+  const [play] = useSound(notificationSound, {
+    // `interrupt` ensures that if the sound starts again before it's
+    // ended, it will truncate it. Otherwise, the sound can overlap.
+    interrupt: true,
+  });
+
+  const [privateChatMessagesState, setPrivateChatMessagesState] = useState(
+    privateChatMessages
+  );
+
+  const chatWithUser =
+    chatSettings.openedChatType === ChatTypes.PRIVATE_CHAT
+      ? chatSettings.recipientId
+      : undefined;
+
+  useEffect(() => {
+    const newMessage = privateChatMessages?.filter(
+      ({ id: id1 }) =>
+        !privateChatMessagesState?.some(({ id: id2 }) => id2 === id1)
+    );
```

Comment:
Instead of `id1` `id2` can we give these semantically meaningful names please.

Maybe something like `existingChatMessageId` and `newPrivateChatMessageId` or similar.

Reply:
Looking at the code later on, you only ever seem to be accessing `newMessage` as `newMessage[0]`, implying that you're only ever checking the first message returned, which makes either that logic wrong, or it pointless to use `.filter` here (instead of `.some` or similar)

Reply:
[FYI] My suggested changes in https://github.com/sparkletown/sparkle/pull/1262#discussion_r604633362 likely override my comments/suggestions made above.